theano/gof/type.py

```diff
@@ -633,10 +669,22 @@ def __hash__(self):
         return hash((type(self), self.ctype, self.freefunc))
 
     def filter(self, data, strict=False, allow_downcast=None):
-        if data is not None and not isinstance(data, self._cdata_type):
-            raise TypeError("expected None or PyCObject/PyCapsule")
+        if data is not None and not isinstance(data, _cdata_type):
+            raise TypeError("expected None or a PyCapsule")
         return data
 
+    def _get_func(self):
+        from theano.scalar import get_scalar_type
+        import theano
```

Comment:
I really wanted to avoid circular imports and this is the definition of Type so I don't think we can move the other one, but for "import theano" sure.